### PR TITLE
Update collections and selection guides for React Aria Components

### PR DESCRIPTION
**Pull request by devongovett**
React Aria Components has a slightly different collection api than the hooks and RSP components. It uses `id` instead of `key`, and has individual item components such as `ListBoxItem` instead of sharing `Item` between all collections. The Collections and Selection guides in React Stately need to continue showing the older API for now (until we extract the new collections out of RAC), so I've copied them into new pages in React Aria and updated them for the new API. Hooks and RSP will continue to link to the stately versions for now.

You can compare the new version with the old one, but I've basically updated the examples, and changed some language to refer to ids instead of keys. I also removed a couple sections that weren't relevant. For example, RAC doesn't support infinite scrolling out of the box yet, and the selected key data type section doesn't apply anymore because we don't use React keys (which convert numbers to strings).

I added a Hooks section at the bottom to link to the stately version for those using React Aria hooks instead of components. Hopefully it's not too confusing.

### CODE REVIEW

File: packages/dev/docs/pages/react-aria/collections.mdx [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+```
+
+As you can see, the items are passed to the `items` prop of the top-level component, which iterates over each
+item and calls the function passed as children to the component. The item object is passed to the function, which
+returns a `<ListBoxItem>`.
+
+### Unique ids
+
+All items in a collection must have a unique id, which is used to determine what items in the collection changed
+when updates occur. By default, React Aria looks for an `id` or `key` property on each item object, which is often returned
```
Comment (snowystinger) [inline packages/dev/docs/pages/react-aria/collections.mdx:105]: remove key?

**Review by reidbarber — DISMISSED** [review]

File: packages/dev/docs/pages/react-aria/collections.mdx [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+</Select>
+```
+
+Note that `useAsyncList` is a convenience hook, not a requirement. You can use any state management or data loading
+library to manage collection items.
+
+## Hooks
+
+If you're using React Aria and React Stately hooks rather than components, the collection API is slightly different.
+See the [React Stately collections](../react-stately/collections.html) documentation for more details.
```
Comment (LFDanLu) [inline packages/dev/docs/pages/react-aria/collections.mdx:315]: mega nitpick: I feel like this disclaimer should be higher up in the page, it isn't immediately obvious that the "hooks" section in the table of contents is referring to our aria/stately hooks nor that this page is RAC specific.

File: packages/dev/docs/pages/react-aria/collections.mdx (hunk 1 above)
Comment (devongovett) [reply packages/dev/docs/pages/react-aria/collections.mdx:105]: I think this is still correct. It's referring to the item objects, not the elements (e.g. `<ListBoxItem>`). We still look for both `id` and `key` as a property. https://github.com/adobe/react-spectrum/blob/9d8c74a344c9b7539f00fa58d0a02b062b6e830e/packages/react-aria-components/src/Collection.tsx#L665

Perhaps we can clarify this. Or do we want to keep supporting it?

File: packages/dev/docs/pages/react-aria/collections.mdx (hunk 2 above)
Comment (devongovett) [reply packages/dev/docs/pages/react-aria/collections.mdx:315]: We are emphasizing the components on all of the pages now, and a few have a hooks section at the bottom. If this turns out to be confusing for people, we can always update it.

**Review by LFDanLu — APPROVED** [review]

**Review by snowystinger — DISMISSED** [review]

**Review by snowystinger — APPROVED** [review]

**Review by reidbarber — APPROVED** [review]

**Pull request merged**